sysvabi64/sysvabi64.rst

```diff
@@ -0,0 +1,1188 @@
+  | ``tprel_hi12``        | ``add``     | R_AARCH64_TLSLE_ADD_TPREL_HI12        |
+  +-----------------------+-------------+---------------------------------------+
+  | ``tprel_lo12_nc``     | ``add``     | R_AARCH64_TLSLE_ADD_TPREL_LO12_NC     |
+  +-----------------------+-------------+---------------------------------------+
+  | ``tprel_lo12_nc``     | ``ldr``     | R_AARCH64_TLSLE_LDST64_TPREL_LO12_NC  |
+  +-----------------------+-------------+---------------------------------------+
+  | ``tprel_g2``          | ``mov[nz]`` | R_AARCH64_TLSLE_MOVW_TPREL_G2         |
+  +-----------------------+-------------+---------------------------------------+
+  | ``tprel_g1``          | ``mov[nz]`` | R_AARCH64_TLSLE_MOVW_TPREL_G1         |
+  +-----------------------+-------------+---------------------------------------+
+  | ``tprel_g1_nc``       | ``movk``    | R_AARCH64_TLSLE_MOVW_TPREL_G1_NC      |
+  +-----------------------+-------------+---------------------------------------+
+  | ``tprel_g0``          | ``mov[nz]`` | R_AARCH64_TLSLE_MOVW_TPREL_G0         |
+  +-----------------------+-------------+---------------------------------------+
+  | ``tprel_g0_nc``       | ``movk``    | R_AARCH64_TLSLE_MOVW_TPREL_G0_NC      |
+  +-----------------------+-------------+---------------------------------------+
+
+.. note::
+
+   Relocations are defined in AAELF64_.
+
+Memory Models
+-------------
+
+The AArch64 A64 instruction set has a number of features and
+constraints which make it desirable to use different code models for
+different sizes of executable or dynamic shared object, to improve
+performance and reduce static code size. The relevant constraints are:
+
+* The LDR (literal) and ADR instructions generate a PC-relative address
+  with a range of +/- 1MiB.
+
+* The ADRP instruction generates a PC-relative, page aligned address
+  with a range of +/- 4GiB (where page = 4KiB).
+
+* The LDR and STR instructions accept a 12-bit unsigned immediate
+  offset, scaled by the access size.
+
+* The B and BL instructiond have a range of +/-128MiB. This is
+  typically used for function / procedure calls.
+
+* Static linkers may insert a veneer (a sequence of instructions) to
+  implement a relocated B or BL to a destination further away than the
+  +/-128MiB range. The size of executable sections must be limited to
+  127 MiB to leave space for veneers to be inserted after the section.
+
+The following code models are defined
+
+.. table::
+
+  +-------+---------+----------------------+-------------------+
+  | Code  | PIC     | Max Segment Size     | Max GOT size      |
+  | Model | support |                      |                   |
+  +=======+=========+======================+===================+
+  | tiny  | yes     | text+got+data < 1MiB | 1MiB              |
+  +-------+---------+----------------------+-------------------+
+  | small | yes     | text+got+data < 4GiB | 32KiB base+offset |
```

Comment:
Also the maximum of text section is 2GB due to use of PREL32 relocations.

Reply:
I've made the change to 2GiB.